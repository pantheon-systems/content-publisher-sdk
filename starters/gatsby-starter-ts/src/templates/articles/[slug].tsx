import { Article } from "@pantheon-systems/pcc-react-sdk";
import { ArticleRenderer } from "@pantheon-systems/pcc-react-sdk/components";
import React from "react";
import Layout from "../../components/layout";
import Seo from "../../components/seo";

const getSeoMetadata = (article: Article) => {
  const tags = article.tags && article.tags.length > 0 ? article.tags : [];
  let authors: string[] = [];
  let publishedTime: string | null = null;

  // Collecting data from metadata fields
  Object.entries(article.metadata || {}).forEach(([key, val]: any) => {
    if (key.toLowerCase().trim() === "author" && val) authors = [val];
    else if (key.toLowerCase().trim() === "date" && val.msSinceEpoch)
      publishedTime = new Date(val.msSinceEpoch).toISOString();
  });

  return {
    title: article.title,
    description: "Article hosted using Pantheon Content Cloud",
    tags,
    authors,
    publishedTime,
  };
};

export default function PageTemplate({ pageContext: { article } }) {
  const seoMetadata = getSeoMetadata(article);
  return (
    <Layout>
      <Seo
        title={seoMetadata.title || undefined}
        description={seoMetadata.description}
        tags={seoMetadata.tags}
        authors={seoMetadata.authors}
        date={seoMetadata.publishedTime || undefined}
      />
      <div className="max-w-screen-lg mx-auto mt-16 prose">
        <ArticleRenderer
          article={article}
          renderTitle={(titleElement) => (
            <div>
              <h1 className="text-3xl font-bold md:text-4xl">{titleElement}</h1>

              {article.updatedAt ? (
                <p className="py-2">
                  Last Updated:{" "}
                  {new Date(article.updatedAt).toLocaleDateString("en-US", {
                    year: "numeric",
                    month: "long",
                    day: "numeric",
                  })}
                </p>
              ) : null}

              <hr className="mt-6 mb-8" />
            </div>
          )}
          bodyClassName={undefined}
          containerClassName={undefined}
          headerClassName={undefined}
          smartComponentMap={undefined}
        />
      </div>
    </Layout>
  );
}
