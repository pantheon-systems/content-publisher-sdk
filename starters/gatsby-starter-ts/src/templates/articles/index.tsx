import { navigate } from "gatsby";
import React from "react";
import { PageGrid } from "../../components/grid";
import Layout from "../../components/layout";
import PageHeader from "../../components/page-header";
import Pagination from "../../components/pagination";

export default function ArticlesListTemplate({
  pageContext: { articles, totalCount, currentPage, pageSize },
}) {
  const RenderCurrentItems = ({ currentItems }) => {
    return <PageGrid data={currentItems} />;
  };

  const onPageChange = (page: number) => {
    if (page === 0) navigate("/articles");
    else navigate(`/articles/${page + 1}`);
  };
  return (
    <Layout>
      <div className="max-w-screen-lg mx-auto">
        <section>
          <PageHeader title="Articles" />
          <RenderCurrentItems currentItems={articles} />
          <div className="mt-4 flex flex-row justify-center items-center">
            <Pagination
              totalCount={totalCount}
              pageSize={pageSize}
              currentPage={currentPage}
              onChange={onPageChange}
            />
          </div>
        </section>
      </div>
    </Layout>
  );
}
