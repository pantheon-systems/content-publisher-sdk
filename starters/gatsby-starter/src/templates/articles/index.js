import { navigate } from "gatsby";
import React from "react";
import { PageGrid } from "../../components/grid";
import Layout from "../../components/layout";
import PageHeader from "../../components/page-header";
import Pagination from "../../components/pagination";

export default function ArticlesListTemplate({
  pageContext: { articles, totalCount, currentPage, pageSize },
}) {
  const RenderCurrentItems = ({ currentItems }) => {
    return <PageGrid contentType="pages" data={currentItems} />;
  };

  const onPageChange = (page) => {
    if (page === 0) navigate("/articles");
    else navigate(`/articles/${page + 1}`);
  };

  return (
    <Layout>
      <div className="max-w-screen-lg mx-auto">
        <section>
          <PageHeader title="Articles" />
          <div className="mt-4 flex flex-row justify-center items-center">
            <Pagination
              totalCount={totalCount}
              pageSize={pageSize}
              currentPage={currentPage}
              onChange={onPageChange}
            />
          </div>
          <RenderCurrentItems currentItems={articles} />
        </section>
      </div>
    </Layout>
  );
}
