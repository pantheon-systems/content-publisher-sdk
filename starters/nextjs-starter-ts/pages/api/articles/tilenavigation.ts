import { NextApiRequest, NextApiResponse } from "next";
import { PCCConvenienceFunctions } from "@pantheon-systems/pcc-react-sdk";

export default async function handler(req: NextApiRequest, res: NextApiResponse) {
    // Check if it is a GET request
    if (req.method !== "GET") {
        return res.status(405).json({ message: "Method not allowed" });
    }

    // Get the document IDs as a comma separated string from the query parameters
    const documentIds = req.query.documentIds as string;

    // Split the document IDs into an array
    const docIds = documentIds.split(",");
    
    // Fetch the article by ID from PCCConvenienceFunctions in parallel
    const articles = await Promise.all(docIds.map(id => PCCConvenienceFunctions.getArticleBySlugOrId(id, "PRODUCTION")));

    // Filter out any articles that are not found
    const foundArticles = articles.filter(article => article !== null);

    // Map each of the article to an object with just the ID and metadata?.image
    const requiredArticleInfo = foundArticles.map(article => ({   
        id: article.id,
        image: article.metadata?.image,
        title: article.title,
        slug: article.metadata?.slug,
    }));

    // Return the articles as a JSON response
    return res.status(200).json({data: requiredArticleInfo});
    
}