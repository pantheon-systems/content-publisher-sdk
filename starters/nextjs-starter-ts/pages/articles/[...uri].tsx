import {
  PantheonProvider,
  PCCConvenienceFunctions,
  type Article,
} from "@pantheon-systems/pcc-react-sdk";
import { NextSeo } from "next-seo";
import queryString from "query-string";
import ArticleView from "../../components/article-view";
import Layout from "../../components/layout";
import { getSeoMetadata } from "../../lib/utils";
import { pantheonAPIOptions } from "../api/pantheoncloud/[...command]";

interface ArticlePageProps {
  article: Article;
  grant: string;
}

export default function ArticlePage({ article, grant }: ArticlePageProps) {
  const seoMetadata = getSeoMetadata(article);

  return (
    <PantheonProvider
      client={PCCConvenienceFunctions.buildPantheonClient({
        isClientSide: true,
        pccGrant: grant,
      })}
    >
      <Layout>
        <NextSeo
          title={seoMetadata.title}
          description={seoMetadata.description}
          openGraph={{
            type: "website",
            title: seoMetadata.title,
            description: seoMetadata.description,
            images: seoMetadata.images,
            article: {
              authors: seoMetadata.authors,
              tags: seoMetadata.tags,
              ...(seoMetadata.publishedTime && {
                publishedTime: seoMetadata.publishedTime,
              }),
            },
          }}
        />

        <div className="prose mx-4 mt-16 text-black sm:mx-6 md:mx-auto">
          <ArticleView article={article} />
        </div>
      </Layout>
    </PantheonProvider>
  );
}

export async function getServerSideProps({
  req: { cookies },
  query: { uri, publishingLevel, pccGrant, ...query },
}) {
  const slugOrId = uri[uri.length - 1];
  const grant = pccGrant || cookies["PCC-GRANT"] || null;

  const article = await PCCConvenienceFunctions.getArticleBySlugOrId(
    slugOrId,
    publishingLevel ? publishingLevel.toString().toUpperCase() : "PRODUCTION",
  );

  if (!article) {
    return {
      notFound: true,
    };
  }

  if (
    article.slug?.trim().length &&
    article.slug.toLowerCase() !== slugOrId?.trim().toLowerCase()
  ) {
    // If the article was accessed by the id rather than the slug - then redirect to the canonical
    // link (mostly for SEO purposes than anything else).
    return {
      redirect: {
        destination: queryString.stringifyUrl({
          url: pantheonAPIOptions.resolvePath(article),
          query: { publishingLevel, ...query },
        }),
        permanent: false,
      },
    };
  }

  return {
    props: {
      article,
      grant,
      recommendedArticles: await PCCConvenienceFunctions.getRecommendedArticles(
        article.id,
      ),
    },
  };
}
