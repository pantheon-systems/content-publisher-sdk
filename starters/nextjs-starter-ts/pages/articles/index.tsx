import {
  Article,
  PCCConvenienceFunctions,
} from "@pantheon-systems/pcc-react-sdk";
import { NextSeo } from "next-seo";
import queryString from "query-string";
import ArticleList from "../../components/article-list";
import Layout from "../../components/layout";
import { PAGE_SIZE } from "../../constants";

async function fetchNextPages(cursor?: string | null | undefined) {
  const url = queryString.stringifyUrl({
    url: "/api/utils/paginate",
    query: {
      pageSize: PAGE_SIZE,
      cursor: cursor,
    },
  });

  const response = await fetch(url);
  const { data, cursor: newCursor } = await response.json();
  return {
    data,
    newCursor,
  };
}

interface Props {
  articles: Article[];
  totalCount: number;
  cursor: string;
}

export default function ArticlesListTemplate({
  articles,
  totalCount,
  cursor,
}: Props) {
  return (
    <Layout>
      <NextSeo title="Articles" description="Articles" />

      <ArticleList
        headerText="Articles"
        articles={articles}
        cursor={cursor}
        totalCount={totalCount}
        fetcher={fetchNextPages}
      />
    </Layout>
  );
}

export async function getServerSideProps() {
  const {
    data: articles,
    totalCount,
    cursor,
  } = await PCCConvenienceFunctions.getPaginatedArticles({
    pageSize: PAGE_SIZE,
  });

  return {
    props: {
      articles,
      cursor,
      totalCount,
    },
  };
}
