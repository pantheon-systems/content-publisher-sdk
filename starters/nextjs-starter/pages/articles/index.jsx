import { PCCConvenienceFunctions } from "@pantheon-systems/pcc-react-sdk";
import { NextSeo } from "next-seo";
import { ArticleGrid } from "../../components/grid";
import Layout from "../../components/layout";
import PageHeader from "../../components/page-header";
import Pagination from "../../components/pagination";
import { usePagination } from "../../hooks/usePagination";

const PAGE_SIZE = 20;

export default function ArticlesListTemplate({ articles, totalCount, cursor }) {
  const {
    data: currentArticles,
    onPageChange,
    fetching,
    currentPage,
  } = usePagination({
    cursor,
    initialArticles: articles,
    pageSize: PAGE_SIZE,
  });

  return (
    <Layout>
      <NextSeo title="Articles" description="Articles" />

      <section className="max-w-screen-3xl mx-auto px-4 pt-16 sm:w-4/5 md:w-3/4 lg:w-4/5 2xl:w-3/4">
        <PageHeader title="Articles" />

        <ArticleGrid articles={currentArticles} />

        <div className="mt-4 flex flex-row items-center justify-center">
          <Pagination
            totalCount={totalCount}
            pageSize={PAGE_SIZE}
            currentPage={currentPage}
            onChange={onPageChange}
            disabled={fetching}
          />
        </div>
      </section>
    </Layout>
  );
}

export async function getServerSideProps() {
  const {
    data: articles,
    totalCount,
    cursor,
  } = await PCCConvenienceFunctions.getPaginatedArticles({
    pageSize: PAGE_SIZE,
  });

  return {
    props: {
      articles,
      totalCount,
      cursor,
    },
  };
}
