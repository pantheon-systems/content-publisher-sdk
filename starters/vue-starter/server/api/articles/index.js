import { PCCConvenienceFunctions } from "@pantheon-systems/pcc-vue-sdk";

export default defineEventHandler(async (event) => {
  const query = getQuery(event);
  let { pageSize, cursor } = query;
  return await PCCConvenienceFunctions.getPaginatedArticles(
    {
      publishingLevel: "PRODUCTION",
      ...(pageSize && { pageSize: pageSize }),
      ...(cursor && { cursor: cursor }),
    },
    {
      publishStatus: "published",
    },
  );
});
