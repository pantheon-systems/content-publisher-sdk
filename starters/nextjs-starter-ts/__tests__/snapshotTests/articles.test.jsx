import { render } from "@testing-library/react";
import ArticlesListTemplate from "../../pages/articles/index";
import ArticlePageTemplate from "../../pages/articles/[...uri]";

import articles from "../data/articles.json";
import article from "../data/article.json";

vi.mock("next/image");
vi.mock("next/router", () => ({
  useRouter: () => ({
    locale: "en",
    pathname: "test/path",
    push: vi.fn(),
    query: {
      slug: "/articles/[slug]",
    },
  }),
}));

/**
 * @vitest-environment jsdom
 */

describe("<PostListTemplate />", () => {
  it("should render with posts", () => {
    const { asFragment } = render(<ArticlesListTemplate articles={articles} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
describe("<PostTemplate />", () => {
  it("should render a post", () => {
    const { asFragment } = render(<ArticlePageTemplate article={article} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
