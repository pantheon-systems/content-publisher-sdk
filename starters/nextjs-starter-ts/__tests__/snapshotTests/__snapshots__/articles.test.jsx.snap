// Vitest Snapshot v1

exports[`<PostListTemplate /> > should render with posts 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen max-h-screen min-w-screen max-w-screen flex flex-col overflow-x-hidden"
  >
    <div
      class="ps-my-0 ps-pt-10 ps-px-5 ps-text-xl"
    >
      <nav>
        <ul
          class="ps-flex ps-flex-row ps-flex-wrap sm:ps-flex-nowrap ps-list-none ps-justify-between max-w-screen-lg ps-mx-auto"
        >
          <li
            class="ps-mr-auto"
          >
            <a
              class="ps-font-sans hover:ps-underline"
              href="/"
            >
              üè† Home
            </a>
          </li>
          <li
            class="ps-mx-4"
          >
            <a
              class="ps-font-sans hover:ps-underline"
              href="/articles"
            >
              üìë Articles
            </a>
          </li>
          <li
            class="ps-mx-4"
          >
            <a
              class="ps-font-sans hover:ps-underline"
              href="/examples"
            >
              ‚öõÔ∏è Examples
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <main
      class="mb-auto"
    >
      <div
        class="max-w-screen-lg mx-auto"
      >
        <section>
          <header
            class="prose text-2xl mx-auto mt-20"
          >
            <h1
              class="text-center mx-auto"
            >
              Articles
            </h1>
          </header>
          <div
            class="ps-max-w-full"
          >
            <section>
              <div
                class="ps-mt-12 ps-grid ps-gap-5 ps-max-w-content ps-mx-auto lg:ps-max-w-screen-lg lg:ps-grid-cols-3"
              >
                <a
                  href="/articles/1"
                >
                  <div
                    class="flex flex-col rounded-lg shadow-lg overflow-hidden cursor-pointer border-2 h-full hover:border-indigo-500"
                  >
                    <div
                      class="flex-shrink-0 relative h-40"
                    >
                      <div
                        class="w-full h-full bg-gradient-to-b from-blue-100 to-blue-500"
                      />
                    </div>
                    <h2
                      class="my-4 mx-6 text-xl leading-7 font-semibold text-gray-900"
                    >
                      Test 1 ‚Üí
                    </h2>
                  </div>
                </a>
                <a
                  href="/articles/1"
                >
                  <div
                    class="flex flex-col rounded-lg shadow-lg overflow-hidden cursor-pointer border-2 h-full hover:border-indigo-500"
                  >
                    <div
                      class="flex-shrink-0 relative h-40"
                    >
                      <div
                        class="w-full h-full bg-gradient-to-b from-blue-100 to-blue-500"
                      />
                    </div>
                    <h2
                      class="my-4 mx-6 text-xl leading-7 font-semibold text-gray-900"
                    >
                      Test 2 ‚Üí
                    </h2>
                  </div>
                </a>
              </div>
            </section>
            <div
              class="ps-sticky lg:ps-bottom-12 ps-bottom-4 ps-mt-10"
            />
          </div>
        </section>
      </div>
    </main>
    <footer
      class="ps-w-full ps-text-white ps-bg-black ps-p-4 ps-mt-12"
    >
      <nav
        class="ps-flex ps-flex-col ps-max-w-lg ps-mx-auto lg:ps-max-w-screen-lg"
      >
        <ul />
      </nav>
      <div
        class="ps-flex ps-my-4 ps-p-2"
      >
        <span
          class="mx-auto"
        >
          ¬© 2023 Built with 
          <a
            class="text-white hover:text-blue-100 underline"
            href="https://nextjs.org/"
          >
            Next.js
          </a>
           and 
          <a
            class="text-blue-500 underline hover:text-blue-100"
            href="https://pantheon.io/"
          >
            Pantheon Content Cloud
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`<PostTemplate /> > should render a post 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen max-h-screen min-w-screen max-w-screen flex flex-col overflow-x-hidden"
  >
    <div
      class="ps-my-0 ps-pt-10 ps-px-5 ps-text-xl"
    >
      <nav>
        <ul
          class="ps-flex ps-flex-row ps-flex-wrap sm:ps-flex-nowrap ps-list-none ps-justify-between max-w-screen-lg ps-mx-auto"
        >
          <li
            class="ps-mr-auto"
          >
            <a
              class="ps-font-sans hover:ps-underline"
              href="/"
            >
              üè† Home
            </a>
          </li>
          <li
            class="ps-mx-4"
          >
            <a
              class="ps-font-sans hover:ps-underline"
              href="/articles"
            >
              üìë Articles
            </a>
          </li>
          <li
            class="ps-mx-4"
          >
            <a
              class="ps-font-sans hover:ps-underline"
              href="/examples"
            >
              ‚öõÔ∏è Examples
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <main
      class="mb-auto"
    >
      <div
        class="max-w-screen-lg mx-auto prose mt-16"
      >
        <div>
          <div>
            <div>
              <h1
                class="text-3xl font-bold md:text-4xl"
              >
                <span>
                  This is a title, there are many like it, but this one is mine
                </span>
              </h1>
              <p
                class="py-2"
              >
                Last Updated: January 1, 1970
              </p>
              <hr
                class="mt-6 mb-8"
              />
            </div>
          </div>
          <div>
            <h1>
              <span>
                This is h1
              </span>
            </h1>
            <h2>
              <span>
                This is h2
              </span>
            </h2>
            <h3>
              <span>
                This is h3
              </span>
            </h3>
            <h4>
              <span>
                This is h4
              </span>
            </h4>
            <h5>
              <span>
                This is h5
              </span>
            </h5>
            <h6>
              <span
                style="font-style: italic;"
              >
                This is h6
              </span>
            </h6>
            <p>
              <span />
            </p>
            <p>
              <span
                style="font-weight: bold;"
              >
                This is bold
              </span>
            </p>
            <p>
              <span
                style="font-style: italic;"
              >
                This is italic
              </span>
            </p>
            <p>
              <span
                style="text-decoration: underline;"
              >
                This is underlined
              </span>
            </p>
            <p>
              <span
                style="text-decoration: underline;"
              />
            </p>
            <p>
              <span
                style="vertical-align: super;"
              >
                aaa
              </span>
              <span
                style="font-weight: bold; font-style: italic; vertical-align: super; text-decoration: underline;"
              >
                This is bold, underlined, italic, strikethrough and supertext
              </span>
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              >
                eee
              </span>
              <span
                style="font-weight: bold; font-style: italic; vertical-align: sub; text-decoration: underline;"
              >
                This is bold, underlined, italic, strikethrough and subscript
              </span>
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              />
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              >
                horizontal line below:
              </span>
            </p>
            <hr />
            <p>
              <span
                style="vertical-align: sub;"
              />
            </p>
            <p>
              <span />
            </p>
            <p>
              <span>
                x
              </span>
              <span
                style="vertical-align: super;"
              >
                3
              </span>
            </p>
            <p>
              <span
                style="vertical-align: super;"
              />
            </p>
            <p>
              <span
                style="vertical-align: super;"
              >
                H
              </span>
              <span
                style="vertical-align: sub;"
              >
                2
              </span>
              <span
                style="vertical-align: super;"
              >
                O x
              </span>
              <span
                style="font-weight: bold; vertical-align: super;"
              >
                o
              </span>
            </p>
            <p>
              <span />
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              >
                Unordered ‚Äúbullet point‚Äù list
              </span>
            </p>
            <ul>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Apple
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Pineapple
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Pen 
                </span>
                <span
                  style="font-weight: bold; vertical-align: sub;"
                >
                  pineapple
                </span>
                <span
                  style="vertical-align: sub;"
                >
                  ¬†
                </span>
                <span
                  style="font-style: italic; vertical-align: sub;"
                >
                  apple
                </span>
                <span
                  style="vertical-align: sub;"
                >
                  ¬†
                </span>
                <span
                  style="vertical-align: sub; text-decoration: underline;"
                >
                  pen
                </span>
              </li>
            </ul>
            <p>
              <span
                style="vertical-align: sub; text-decoration: underline;"
              />
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              >
                Unordered list with different symbols
              </span>
            </p>
            <ul>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Duck
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Duck2
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Goose
                </span>
              </li>
            </ul>
            <p>
              <span
                style="vertical-align: sub;"
              />
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              >
                Todo list with checkmarks
              </span>
            </p>
            <ul>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Veni
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Vidi
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Vici
                </span>
              </li>
            </ul>
            <p>
              <span
                style="vertical-align: sub; text-decoration: underline;"
              />
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              >
                Ordered ‚Äúnumbered‚Äù list
              </span>
            </p>
            <ol>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Waking in the morning
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Gotta go downstairs
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Gotta have cereal
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  I see my friends
                </span>
              </li>
              <li>
                <span
                  style="vertical-align: sub;"
                >
                  Gotta get down on Friday
                </span>
              </li>
            </ol>
            <p>
              <span
                style="vertical-align: sub;"
              />
            </p>
            <p>
              <span />
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              />
            </p>
            <p>
              <span
                style="font-weight: bold; vertical-align: sub;"
              >
                A wild table appears
              </span>
            </p>
            <table
              class="text-black"
            >
              <tbody>
                <tr>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        Merged cols 1 & 2
                      </span>
                    </p>
                  </td>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        Col 3
                      </span>
                    </p>
                  </td>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        Col 4
                      </span>
                    </p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        A
                      </span>
                    </p>
                  </td>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        B
                      </span>
                    </p>
                  </td>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        C
                      </span>
                    </p>
                  </td>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        D
                      </span>
                    </p>
                  </td>
                </tr>
                <tr>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        ·Éê
                      </span>
                    </p>
                  </td>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        ·Éë
                      </span>
                    </p>
                  </td>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        ·É°
                      </span>
                    </p>
                  </td>
                  <td>
                    <p>
                      <span
                        style="vertical-align: sub;"
                      >
                        ·Éì
                      </span>
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
            <p>
              <span
                style="vertical-align: sub;"
              />
            </p>
            <p>
              <span
                style="vertical-align: sub;"
              >
                Some more paragraphs after everything else.
              </span>
            </p>
            <p>
              <span />
            </p>
            <p>
              <span />
            </p>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="ps-w-full ps-text-white ps-bg-black ps-p-4 ps-mt-12"
    >
      <nav
        class="ps-flex ps-flex-col ps-max-w-lg ps-mx-auto lg:ps-max-w-screen-lg"
      >
        <ul />
      </nav>
      <div
        class="ps-flex ps-my-4 ps-p-2"
      >
        <span
          class="mx-auto"
        >
          ¬© 2023 Built with 
          <a
            class="text-white hover:text-blue-100 underline"
            href="https://nextjs.org/"
          >
            Next.js
          </a>
           and 
          <a
            class="text-blue-500 underline hover:text-blue-100"
            href="https://pantheon.io/"
          >
            Pantheon Content Cloud
          </a>
        </span>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
